src/bin/pg_autoctl/watch_colspecs.h

```diff
@@ -0,0 +1,173 @@
+/*
+ * src/bin/pg_autoctl/watch.h
+ *     Implementation of a CLI to show events, states, and URI from the
+ *     pg_auto_failover monitor.
+ *
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the PostgreSQL License.
+ *
+ */
+
+#ifndef WATCH_COLSPECSH
+#define WATCH_COLSPECSH
+
+#include "watch.h"
+
+/* Column Specifications, so that we adapt to the actual/current screen size */
+typedef enum
+{
+	COLUMN_TYPE_NAME = 0,
+	COLUMN_TYPE_ID,
+	COLUMN_TYPE_REPLICATION_QUORUM,
+	COLUMN_TYPE_CANDIDATE_PRIORITY,
+	COLUMN_TYPE_HOST_PORT,
+	COLUMN_TYPE_TLI_LSN,
+	COLUMN_TYPE_CONN_HEALTH,
+	COLUMN_TYPE_CONN_HEALTH_LAG,
+	COLUMN_TYPE_CONN_REPORT_LAG,
+	COLUMN_TYPE_REPORTED_STATE,
+	COLUMN_TYPE_ASSIGNED_STATE,
+
+	COLUMN_TYPE_LAST
+} ColumnType;
+
+typedef struct ColSpec
+{
+	ColumnType type;
+	char name[NAMEDATALEN];
+	int len;
+} ColSpec;
+
+#define MAX_COL_SPECS 12
+
+typedef struct ColPolicy
+{
+	char name[NAMEDATALEN];
+	int totalSize;
+	ColSpec specs[MAX_COL_SPECS];
+} ColPolicy;
+
+/*
+ * A column policy is a list of column specifications.
+ *
+ * We have a static list of policies, and we pick one at run-time depending on
+ * the current size of the terminal window and depending on the actual data
+ * size to be displayed, which is also dynamic.
+ */
+ColPolicy ColumnPolicies[] = {
+	{
+		"minimal",
+		0,
+		{
+			{ COLUMN_TYPE_ID, "Id", 0 },
+			{ COLUMN_TYPE_REPORTED_STATE, "Reported State", 0 },
+			{ COLUMN_TYPE_ASSIGNED_STATE, "Assigned State", 0 },
+			{ COLUMN_TYPE_LAST, "", 0 }
+		}
+	},
+	{
+		"very terse",
+		0,
+		{
+			{ COLUMN_TYPE_ID, "Id", 0 },
+			{ COLUMN_TYPE_CONN_REPORT_LAG, "Lag(R)", 0 },
+			{ COLUMN_TYPE_REPORTED_STATE, "Reported State", 0 },
+			{ COLUMN_TYPE_ASSIGNED_STATE, "Assigned State", 0 },
+			{ COLUMN_TYPE_LAST, "", 0 }
+		}
+	},
+	{
+		"terse",
+		0,
+		{
+			{ COLUMN_TYPE_ID, "Id", 0 },
```

Comment:
I think Name is more important to the users than Id (same for very terse and minimal)
```suggestion
			{ COLUMN_TYPE_ID, "Name", 0 },
```

Reply:
I edited that now. The reason I picked the Id over the Name is that it's very terse indeed, allowing for more space for other information. That said, I suppose users will want to see the names of the nodes rather than their id in most situations.